src/java-runtime/java/mono/android/MonoPackageManager.java

```diff
@@ -162,4 +166,31 @@ public static String[] getDependencies ()
 	{
 		return MonoPackageManager_Resources.Dependencies;
 	}
+
+	static byte[] getMappingXml (Context context)
+	{
+		try {
+			AssetManager manager = context.getAssets();
+			String[] assets = manager.list ("xa-internal");
+			if (assets == null) {
+				return null;
+			}
+			for (String asset: assets) {
+				if (!asset.equals ("xa-mam-mapping.xml")) {
+					continue;
+				}
+				Log.d ("*jonp*", "# jonp: found `xa-mam-mapping.xml`; trying to load...");
+				InputStream s = manager.open ("xa-internal/xa-mam-mapping.xml");
+				Log.d ("*jonp*", "# jonp: s? " + (s != null));
+				byte[] contents = new byte[s.available ()];
+				int r = s.read (contents);
+				Log.d ("*jonp*", "# jonp: read " + r + " bytes from inputstream! expected " + contents.length + "!");
+				s.close ();
```

Comment:
Java doesn't have the C# `using` block, so is one supposed to call `close()` in a `try`-`finally` block?

Reply:
Java now has a "try-with-resources" construct, akin to C# `using` blocks.  Code updated to use that.